drivers/ieee802154/ieee802154_dw1000.c

```diff
@@ -1364,14 +1363,14 @@ static int dwt_configure_rf_phy(const struct device *dev)
 
 	if (IS_ENABLED(CONFIG_ENABLE_MANUAL_TX_POWER_CONTROL)) {
 		sys_cfg |= DWT_SYS_CFG_DIS_STXP;
-		if (rf_cfg->prf == DWT_PRF_64M) {
+		if (IS_ENABLED(CONFIG_PRF_64MHZ)) {
```

Comment:
You change breaks configuration per instance, what TBH is not supported by the driver completely but the part you change does it. You have to keep rf_cfg->prf and use devicetree property or runtime configuration.

Reply:
@jfischer-no I agree that iif the driver supported multiple instances then it would have to maintain instance support here. But this is not the case as you say.

Therefore we this is not a clear-cut decision but a trade off: As a user you want to be able to configure all driver-specific parameters in one place. I had explicitly to @loopsmark to introduce a preliminary change that makes the driver instance capable and moves all KConfig variables to DT. But the contributor considered this to be too much work and too far away from his original intent. Therefore, by Zephyr's reviewer rules, I did not insist to add extra requirements to this PR. My conscious decision was that if I have to choose between exposing the user to an inconsistent configuration API versus getting no contribution at all, we rather leave the multi-instance migration to a a future PR by some contributor that actually needs it.

BTW: Similar decisions are regularly being approved in the sub-system, first and foremost in the nRF5 driver. If we ask for DT configuration here, then Nordic should be the first to make their driver multi-instance capable and move all their configuration to DT.

Reply:
Multiinstance support is straightforward, but not after this change. As long as it breaks, my NACK stays.


> BTW: Similar decisions are regularly being approved in the sub-system, first and foremost in the nRF5 driver. If we ask for DT configuration here, then Nordic should be the first to make their driver multi-instance capable and move all their configuration to DT.

Perhaps, but there is a nuance: you can have multiple DW1000 transceivers connected to a SoC, but not multiple nRF5 transceivers. Your statement is still a little confusing, what does Nordic have to do with this PR?

Reply:
@jfischer-no I follow your argument, that multiple instances make sense for external transceivers but not for nRF5 radios. But my argument re providing a consistent configuration interface to application developers and not extending the scope of PRs unless the contributor agrees is still valid.

So what I propose is that we re-introduce `rf_cfg->prf` to maintain a straightforward migration path towards multi-instance capability. But still we make all new parameters configurable via KConfig until someone else steps up and actually wants to make the driver multi-instance capable. We do not force the contributor to move all configuration variables to DT.

Reply:
This is fine as long as the drivers are EXPERIMENTAL/UNSTABLE.